tests/unit/acquisition/test_function.py

```diff
@@ -496,4 +502,73 @@ def _build_with_sampler(
             return raise_
 
     data = Dataset(tf.zeros([0, 1]), tf.zeros([0, 2]))
-    _Acq(100).prepare_acquisition_function(data, _ArbitraryDimTwoOutputModel())
+    _Acq(100).prepare_acquisition_function(data, _IndOnlyDimTwoOutputModel())
+
+
+@pytest.mark.parametrize("sample_size", [0, -2])
+def test_batch_reparametrization_sampler_raises_for_invalid_sample_size(sample_size: int) -> None:
+    with pytest.raises((ValueError, tf.errors.InvalidArgumentError)):
+        BatchReparametrizationSampler(sample_size, QuadraticWithUnitRBF())
+
+
+# todo this one uses one output dim, while we should test with more than one
```

Comment:
i've got a draft fix that tests this with two output dims. The mean is 0.8% off (which is too high), but the covariance is correct. I'm not sure whether to add this to the PR yet. It would mean testing the utility model in more depth

Reply:
i'll probably do this, it's not that much work

Reply:
it would also mean `_IndOnlyDimTwoOutputModel` above could be replaced, which would give us consistency across the tests